### ISSUE TITLE
ToggleSwitch 在Android开启无障碍朗读屏幕后报错崩溃

### ISSUE THREAD

**Issue by SunnyDesignor** [AvaloniaUI/Avalonia#20354]
### Describe the bug

页面中含有 ToggleSwitch 的项目在Android设备上开启无障碍朗读屏幕后抛出以下异常：

System.Reflection.TargetException:“Object type AndroidX.Core.View.Accessibility.AccessibilityNodeInfoCompat does not match target type Avalonia.Android.Automation.ToggleNodeInfoProvider.”

“System.Reflection.TargetException”类型的未经处理的异常在 Mono.Android.dll 中发生
Object type AndroidX.Core.View.Accessibility.AccessibilityNodeInfoCompat does not match target type Avalonia.Android.Automation.ToggleNodeInfoProvider.

除了ToggleSwitch 外似乎还有其他控件也存在这个问题。

### To Reproduce

创建一个avalonia测试项目，在页面中添加一个ToggleSwitch，然后在 Android 手机上启用TalkBack，并编译运行测试项目APP，打开APP后，它在特定设备上一定会报错，详细环境见Additional context

### Expected behavior

不发生任何异常，并且TalkBack能正常聚焦和朗读ToggleSwitch 

### Avalonia version

11.3.10

### OS

Android

### Additional context

使用的无障碍读屏：TalkBack
安卓版本：Android 16
Avalonia 版本：11.3.10
手机厂商：Vivo

**Comment by Daxter98** [comment]
Thanks for opening the issue here! 

I'll add a bit more info: this problem seems to be specific to certain devices. I tried this on a Samsung S22 running Android 15, and there was no crash. However, a Motorola Edge 50 Pro running Android 16 crashed immediately after I tried to input something. Only one emulator, running Android 16 with Talkback enabled, caused the crash.

On both physical devices, no accessibility configurations were turned on. I tried using both a toggle switch and a TextBox with RevealPassword.

```
<TextBox Watermark="Password"
         PasswordChar="*"
         VerticalAlignment="Center"
         HorizontalAlignment="Stretch"
         Classes="revealPasswordButton"/>
```

**Comment by timunie** [comment]
I wonder if nightly builds have the same issue. Could you test it?

**Comment by Daxter98** [comment]
I just tested on latest CI build (12.0.999-cibuild0061063-alpha). The reported error is not present there, but I had to use a custom font for the app to execute.

If I only leave the .WithInterFont() method in the builder, the app would crash on startup:
```
System.InvalidOperationException: Could not create glyphTypeface. Font family: $Default (key: ). Style: Normal. Weight: Normal. Stretch: Normal
   at Avalonia.Media.Typeface.get_GlyphTypeface()
   at Avalonia.Rendering.Composition.Compositor.get_DiagnosticTextRenderer()
   at Avalonia.Rendering.Composition.Compositor.CreateCompositionTarget(Func`1 surfaces)
   at Avalonia.Rendering.Composition.CompositingRenderer..ctor(IRenderRoot root, Compositor compositor, Func`1 surfaces)
   at Avalonia.Controls.TopLevel..ctor(ITopLevelImpl impl, IAvaloniaDependencyResolver dependencyResolver)
   at Avalonia.Controls.TopLevel..ctor(ITopLevelImpl impl)
   at Avalonia.Controls.Embedding.EmbeddableControlRoot..ctor(ITopLevelImpl impl)
   at Avalonia.Android.AvaloniaView..ctor(Context context)
   at Avalonia.Android.AvaloniaMainActivity.InitializeAvaloniaView(Object initialContent)
   at Avalonia.Android.AvaloniaActivity.OnCreate(Bundle savedInstanceState)
   at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_savedInstanceState)
```

With the following code it works:
```
[Application]
public class MainApplication : AvaloniaAndroidApplication<App>
{
    public MainApplication(IntPtr handle, JniHandleOwnership transfer)
        : base(handle, transfer)
    {
    }

    protected override AppBuilder CustomizeAppBuilder(AppBuilder builder)
    {
        return base.CustomizeAppBuilder(builder)
            .With(new FontManagerOptions
            {
                DefaultFamilyName = "avares://Apb.Ibero.Visitantes.Avalonia/Assets/Fonts/Roboto-Regular.ttf#Roboto"
            })
            .UseReactiveUI();
    }
}
```
Also, the clear text button added via the clearButton class doesn't work anymore. It only closes and reopens the input. However, that's another issue.

**Comment by SineVector241** [comment]
Having the same issue with some users reporting the same crash. Except I am using 11.3.9.

![Image](https://github.com/user-attachments/assets/ae07d508-6782-407c-95b5-4fa53fc8d555)

**Comment by SunnyDesignor** [comment]
As of version 11.3.11, this bug has not been fixed in my tests

**Comment by rabbitism** [comment]
> As of version 11.3.11, this bug has not been fixed in my tests

Is there any pr trying to fix this issue?

**Comment by SunnyDesignor** [comment]
I tested the nightly build `12.0.999-cibuild0061483-alpha`. While it indeed fixed the crash, the accessibility behavior is still defective:
﻿
Swipe Navigation: Works correctly. The TextBlock and ToggleSwitch can be focused by swiping.
Explore by Touch: Broken. Only the Button gets focus when touched directly. Touching the TextBlock or ToggleSwitch only triggers vibration feedback but fails to focus or read out the content.
﻿
demo layout :
```xml
<Grid>
    <StackPanel VerticalAlignment="Center">
        <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center" 
           Focusable="True" AutomationProperties.Name="My TextBlock"/>
        <ToggleSwitch HorizontalAlignment="Center" 
             AutomationProperties.
Name="My Switch"
             AutomationProperties.
HelpText="Double tap to toggle" 
             Focusable="True"/>
        <Button HorizontalAlignment="Center" Content="My Button"/>
    </StackPanel>
</Grid>
```
﻿
Since I cannot use alpha builds in production, I tested the latest release 11.3.11 (which mentions fixing #20438), but it did not resolve this issue, app still crashes on 11.3.11

**Comment by SunnyDesignor** [comment]
I have tested 11.3.12, but the issue remains unfixed. Currently, I am circumventing it by overriding OnCreateAutomationPeer

**Comment by rabbitism** [comment]
> I have tested 11.3.12, but the issue remains unfixed. Currently, I am circumventing it by overriding OnCreateAutomationPeer

What is the acceptable automation peer?

**Comment by SunnyDesignor** [comment]
I have currently implemented temporary custom controls and replaced all CheckBox and ToggleSwitch controls with custom controls to avoid app crashes when TalkBack is enabled.

```c#
    public class ToggleSwitch : Avalonia.Controls.ToggleSwitch
    {
        protected override Type StyleKeyOverride => typeof(Avalonia.Controls.ToggleSwitch);

        protected override AutomationPeer OnCreateAutomationPeer()
        {
            return new SafeSwitchAutomationPeer(this);
        }
    }

    public class CheckBox : Avalonia.Controls.CheckBox
    {
        protected override Type StyleKeyOverride => typeof(Avalonia.Controls.CheckBox);

        protected override AutomationPeer OnCreateAutomationPeer()
        {
            return new SafeSwitchAutomationPeer(this);
        }
    }

    public class SafeSwitchAutomationPeer : ControlAutomationPeer
    {
        public SafeSwitchAutomationPeer(Control owner) : base(owner)
        {
        }

        protected override AutomationControlType GetAutomationControlTypeCore()
        {
            return AutomationControlType.CheckBox;
        }

        protected override string GetClassNameCore()
        {
            return "SafeSwitch";
        }
    }
```

**Comment by timunie** [comment]
Why not sending a PR with your fixes?
